pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -274,8 +274,14 @@ type DiscordConfig struct {
 	// The secret's key that contains the Discord webhook URL.
 	// The secret needs to be in the same namespace as the AlertmanagerConfig
 	// object and accessible by the Prometheus Operator.
-	// +required
-	APIURL v1.SecretKeySelector `json:"apiURL"`
```

Comment:
Should we add it back?

Reply:
Hmmm... No rollback is required here. Users using the old version will still use `apiURL`. New users need to either `apiURL` or `webhookURLFile` is required.

Reply:
Hmmm! Not sure, maybe I'm missing something.
but from my understanding you should keep both and see what users is setting, the apiURL now as optional.

Also log a deprecation warning or maybe check what's the alertmanager version to allow one or other.

Reply:
> Hmmm! Not sure, maybe I'm missing something. but from my understanding you should keep both and see what users is setting, the apiURL now as optional.
> 
> Also log a deprecation warning or maybe check what's the alertmanager version to allow one or other.

Here：
https://github.com/prometheus-operator/prometheus-operator/blob/8df0ff1baf8e003f18ca224e58306a1c28ca5d16/pkg/alertmanager/amcfg.go#L2171-L2173

Just keep the CRD `apiURL` field unchanged, the internal structure is passed using `webhookUrl`.
In a future version, add the `apiURL` deprecated `CHANGE` type PR.